temporal-sdk/src/test/java/io/temporal/workflow/GetCronScheduleFromWorkflowInfoTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.workflow;
+
+import static io.temporal.testing.internal.SDKTestOptions.newWorkflowOptionsWithTimeouts;
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import io.temporal.client.WorkflowFailedException;
+import io.temporal.client.WorkflowStub;
+import io.temporal.failure.CanceledFailure;
+import io.temporal.testing.internal.SDKTestWorkflowRule;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.Assume;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+
+public class GetCronScheduleFromWorkflowInfoTest {
+
+  @Rule public TestName testName = new TestName();
+
+  @Rule
+  public SDKTestWorkflowRule testWorkflowRule =
+      SDKTestWorkflowRule.newBuilder()
+          .setWorkflowTypes(TestGetCronScheduleWorkflowsFuncImpl.class)
+          .build();
+
+  @Test
+  public void testGetCronScheduleFromWorkflowInfo() throws InterruptedException {
+    Assume.assumeFalse("skipping for docker tests", testWorkflowRule.isUseExternalService());
+
+    WorkflowStub workflowStub =
+        testWorkflowRule
+            .getWorkflowClient()
+            .newUntypedWorkflowStub(
+                "TestGetCronScheduleWorkflowsFunc",
+                newWorkflowOptionsWithTimeouts(testWorkflowRule.getTaskQueue())
+                    .toBuilder()
+                    .setCronSchedule("0 0 * * *")
+                    .build());
+    testWorkflowRule.registerDelayedCallback(Duration.ofDays(1), workflowStub::cancel);
```

Comment:
Why do we need it here?

Reply:
to hit the cron at least once, and then cancel it after a day (to stop it)

Reply:
the cron is set for "At 00:00" and we cancel it after a day to make sure it invokes the workflow method at least once when its tested. there seems to be some issues with testing cron workflows in general that will take a look at next.

Reply:
Why do we need to cancel it at all after that day? I'm just not sure how it affects the test and why it's important for it. If the test can be implemented without this cancellation - it should (for example, by canceling from inside the worklfow?). Because you likely will face troubles adapting this test for a dockerized temporal if this is an important piece for the test.